examples/cpp/customserializable/main.cpp

```diff
@@ -0,0 +1,133 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <iostream>
+#include <sstream>
+
+#include <geode/CacheFactory.hpp>
+#include <geode/PoolManager.hpp>
+#include <geode/PdxWrapper.hpp>
+
+using namespace apache::geode::client;
+
+class Order : public PdxSerializable {
+ public:
+  Order() {
+    orderID = (unsigned long)0;
+    quantity = (unsigned int)0;
+    name = "";
+  };
+
+  Order(unsigned long inOrderID, const std::string& inName,
```

Comment:
Additionally consider adding another constructor that accepts an rvalue. This allows the caller to avoid copy construction on std::string.
```
Order(unit32_t orderID, std::string&& name, uint32_t quantity)
    : orderID(orderID), name(std::move(name)), quantity(quantity) {}
```

Reply:
Another option is universal reference and std::forward.  Then you only need 1 constructor.  

```cpp
template <typename T>
Order(uint32_t orderId, T&& name, uint32_t quantity)
  : orderId(orderid), name(std::forward(name), quantity(quantity) {}
```

Reply:
We ended up introducing the rvalue reference constructor to keep things somewhat simple in the example.

Reply:
I like @dgkimura solution even better!